M/mlpack/build_tarballs.jl

```diff
@@ -0,0 +1,177 @@
+version = v"3.3.0-a1"
+sources = [
+    # Current git master branch as of 12/20/2019.
+    # This will be replaced with the actual mlpack 3.3.0 release... but that
+    # release is dependent on coherent Julia support, so this has to come
+    # first...
+    "https://www.ratml.org/misc/mlpack-3.3.0-a1.tar.gz" =>
+    "70b386c191465feff93d63ac299612e93eb943ea9144525108674ada037321cf"
+]
+
+script = raw"""
+cd ${WORKSPACE}/srcdir/mlpack-*/
+
+mkdir build && cd build
+
+# In order to convince mlpack to build Julia bindings, we have to use CMake
+# to specify the location of the Julia program.  But... it turns out that
+# all that CMake needs is some kind of executable program that prints the
+# version.  So we'll just create a crappy little script, since Julia may not
+# be available in the build environment...
+echo "#!/bin/bash" > julia
+echo "echo \"Fake Julia version 1.3.0\"" >> julia
+chmod +x julia
+
+FLAGS=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}
+       -DCMAKE_INSTALL_PREFIX=${prefix}
+       -DBUILD_SHARED_LIBS=ON
+       -DDEBUG=OFF
+       -DPROFILE=OFF
+       -DUSE_OPENMP=OFF
+       -DBoost_NO_BOOST_CMAKE=1
+       -DBUILD_JULIA_BINDINGS=ON
+       -DJULIA_EXECUTABLE="${PWD}/julia"
+       -DBUILD_CLI_EXECUTABLES=OFF
+       -DBUILD_TESTS=OFF)
+
+if [[ "${nbits}" == 64 ]] && [[ "${target}" != aarch64* ]]; then
+    # We need to rename some functions for compatibility with Julia's OpenBLAS
+    SYMB_DEFS=()
+    for sym in cgbcon cgbsv cgbsvx cgbtrf cgbtrs cgecon cgees cgeev cgeevx cgehrd cgels cgelsd cgemm cgemv cgeqrf cgesdd cgesv cgesvd cgesvx cgetrf cgetri cgetrs cgges cggev cgtsv cgtsvx cheev cheevd cherk clangb clange clanhe clansy cpbtrf cpocon cposv cposvx cpotrf cpotri cpotrs ctrcon ctrsyl ctrtri ctrtrs cungqr dasum ddot dgbcon dgbsv dgbsvx dgbtrf dgbtrs dgecon dgees dgeev dgeevx dgehrd dgels dgelsd dgemm dgemv dgeqrf dgesdd dgesv dgesvd dgesvx dgetrf dgetri dgetrs dgges dggev dgtsv dgtsvx dlahqr dlangb dlange dlansy dlarnv dnrm2 dorgqr dpbtrf dpocon dposv dposvx dpotrf dpotri dpotrs dstedc dsyev dsyevd dsyrk dtrcon dtrevc dtrsyl dtrtri dtrtrs ilaenv sasum sdot sgbcon sgbsv sgbsvx sgbtrf sgbtrs sgecon sgees sgeev sgeevx sgehrd sgels sgelsd sgemm sgemv sgeqrf sgesdd sgesv sgesvd sgesvx sgetrf sgetri sgetrs sgges sggev sgtsv sgtsvx slahqr slangb slange slansy slarnv snrm2 sorgqr spbtrf spocon sposv sposvx spotrf spotri spotrs sstedc ssyev ssyevd ssyrk strcon strevc strsyl strtri strtrs zgbcon zgbsv zgbsvx zgbtrf zgbtrs zgecon zgees zgeev zgeevx zgehrd zgels zgelsd zgemm zgemv zgeqrf zgesdd zgesv zgesvd zgesvx zgetrf zgetri zgetrs zgges zggev zgtsv zgtsvx zheev zheevd zherk zlangb zlange zlanhe zlansy zpbtrf zpocon zposv zposvx zpotrf zpotri zpotrs ztrcon ztrsyl ztrtri ztrtrs zungqr; do
+        SYMB_DEFS+=("-D${sym}=${sym}_64")
+    done
+    export CXXFLAGS="${SYMB_DEFS[@]}"
+fi
+
+if [[ $target == *powerpc* ]]; then
+    FLAGS+=(-DCMAKE_EXE_LINKER_FLAGS="-L/opt/${target}/${target}/lib${arch_ext}/ -lgomp")
```

Comment:
You probably need to link also against `-lpthread` and `-ldl`

Reply:
Yep, `-ldl` was needed also.  I'm not sure exactly why it is that PPC is behaving badly here, but I can't seem to reproduce it locally (hence the fighting with Azure).  If you have any further ideas I'd be happy to take them but I'm also happy to continue semi-guided random modification of the build rules until it works. :smile: